OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs

```diff
@@ -126,6 +126,11 @@ public bool IsEnemyUnit(Actor a)
 				&& !a.GetEnabledTargetTypes().IsEmpty;
 		}
 
+		public bool IsNotHiddenUnit(Actor a)
+		{
+			return !a.TraitsImplementing<IVisibilityModifier>().Any() || a.TraitsImplementing<IVisibilityModifier>().Where(ha => ha.IsVisible(a, Player)).Any();
```

Comment:
This can likely simplify to `return !a.TraitsImplementing<IVisibilityModifier>().Any(ha => ha.IsVisible(a, Player));`

Reply:
That simplifed one is the hardcore bug I just fix in fact (quite time taking for me to understand)  when there is only `!a.TraitsImplementing<IVisibilityModifier>().Any(ha => ha.IsVisible(a, Player));`,  it means only choose those who apply interface `IVisibilityModifier` trait meanwhile is visible.  It will cause bug that make all States apply it receive nothing when scanning.

Reply:
Ah my bad, I missed that the second condition doesn't have a `!`. This should still change to `return !a.TraitsImplementing<IVisibilityModifier>().Any() || a.TraitsImplementing<IVisibilityModifier>().Any(ha => ha.IsVisible(a, Player));` however.